changelog/copy-constructor-priority.dd

```diff
@@ -0,0 +1,30 @@
+Copy constructor priority with regards to postblit
+
+Up until this release, the postblit had priority over the copy constructor,
+no matter if the former was user-defined or generated. This prevented new
+code that uses the copy constructor to interact with legacy library code
+that uses the postblit. To enable this interaction, having a generated postblit
+and a copy constructor (user-defined or generated) is now deprecated. For example:
+
+---
+// library code using postblit
+struct A
+{
+    this(this) {}
+}
+
+// new code using copy constructor
+struct B
+{
+    A a;
+    this(const scope ref B) {}
+}
+---
+
+Will issue a deprecation warning at the definition of `B` stating that the postblit
+is going to be preferred to the copy constructor. To instruct the compiler that
+the copy constructor has to be used instead of the postblit, the developer has
+to disable the postblit for `struct B`.
+
+If `B` has both a user-defined postblit and a copy constructor (generated or user-defined),
+the postblit will continue to have priority.
```

Comment:
I'd rephrase this a bit, saying that the user has two courses of actions to solve this deprecation warning:
- Explicitly add `@disable this(this);`, which instructs the compiler that the user wants the copy ctor to be used over the postblit;
- Remove or disable the copy constructor;
- Explicitly define a postblit;

And hopefully emphasis on the fact that the first solution should be preferred (it's likely to be the right one for most people).

Reply:
The copy constructor can be overloaded so I think that for the ease of implementation we can still trigger the deprecation even if all copy constructors are disabled. You can simply add `this(this) {}` to achieve the same effect as disabling all copy constructors.